pkg/transform/oauth_transform_test.go

```diff
@@ -0,0 +1,305 @@
+package transform
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/fusor/cpma/pkg/transform/oauth"
+	"github.com/fusor/cpma/pkg/transform/secrets"
+	"github.com/stretchr/testify/assert"
+
+	configv1 "github.com/openshift/api/legacyconfig/v1"
+	k8sjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func loadTestIdentityProviders() []oauth.IdentityProvider {
+	// TODO: Something is broken here in a way that it's causing the translaters
+	// to fail. Need some help with creating test identiy providers in a way
+	// that won't crash the translator
+
+	// Build example identity providers, this is straight copy pasted from
+	// oauth test, IMO this loading of example identity providers should be
+	// some shared test helper
+	file := "testdata/bulk-test-master-config.yaml" // File copied into transform pkg testdata
+	content, _ := ioutil.ReadFile(file)
+	serializer := k8sjson.NewYAMLSerializer(k8sjson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	var masterV3 configv1.MasterConfig
+	_, _, _ = serializer.Decode(content, nil, &masterV3)
+
+	var identityProviders []oauth.IdentityProvider
+	for _, identityProvider := range masterV3.OAuthConfig.IdentityProviders {
+		providerJSON, _ := identityProvider.Provider.MarshalJSON()
+		provider := oauth.Provider{}
+		json.Unmarshal(providerJSON, &provider)
```

Comment:
Is this pattern normal for parsing the master-config.yml.
Seemed odd to:
 - Parse YAML to Struct
 - Extract a piece of Struct and marshal to JSON
 - Unmarshal JSON to a Struct

Realize it's a copy/paste from an existing block of test code and not related to testing the ETL pattern.
More just a general question, is there a reason for doing this?

Reply:
I'll have to defer to @eriknelson about the 'normacly' of it.

Reply:
After digging into this I think this is the best way. But we really should put a comment above it to describe WHY we're doing something that looks so weird :)

`identityProvider.Provider` is of type `runtime.RawExtension`. As you can see here https://github.com/openshift/api/blob/master/legacyconfig/v1/types.go#L852

`runtime.RawExtension` has 2 attributes:

```golang
	Raw []byte `protobuf:"bytes,1,opt,name=raw"`
	Object Object `json:"-"`
```
https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/types.go#L94-L102

So in order to get the `Kind`, etc. needed to fill in the `oauth.IdentityProvider`, that `[]byte` needs to be converted into something useful.

That's where the MarshalJSON comes into play.

Unless there exists utility methods in `apimachinery` to do this for us, looks like this seems to be the best way. Also it's a test :)

Reply:
That's exactly why. 
Earlier in the project, we also talked about being able to tap into `apimachinery` to get full support of the `kind` genre.
  
With more coming components to cover it's something it might worth looking into it to save us time redefining the target for each new component wrapped inside each RawExtension. 

Any pointer to reach out for one of the OCP/Component (like Auth) Team's member to do a knowledge transfer to fast forward use of their API?